circe/src/main/scala/org/http4s/circe/CirceInstances.scala

```diff
@@ -141,6 +141,11 @@ trait CirceInstances extends JawnInstances {
   implicit def streamJsonArrayEncoder[F[_]]: EntityEncoder[F, Stream[F, Json]] =
     streamJsonArrayEncoderWithPrinter(defaultPrinter)
 
+  implicit def streamJsonArrayDecoder[F[_]: Sync]: EntityDecoder[F, Stream[F, Json]] =
+    EntityDecoder.byteArrayDecoder[F].map { array =>
+      fs2.byteArrayParser[F].apply(Stream.fromIterator(array.iterator))
+    }
+
```

Comment:
Thanks!  Two things:

1. This is going to decode the entire body to an array, and then restream it.
2. We should be able to do this without the circe-fs2 dependency.  It's a fine library, but we already have jawn-fs2, which is just a more abstract version of it.

How about this:

```suggestion
  implicit def streamJsonArrayDecoder[F[_]: Sync]: EntityDecoder[F, Stream[F, Json]] =
    EntityDecoder.decodeBy(MediaType.application.json) { media =>
      DecodeResult.successT(media.body.chunks.through(jawnfs2.unwrapJsonArray))
    }
```

Reply:
> This is going to decode the entire body to an array, and then restream it.

whoops I got so preoccupied with following the types that I forgot about this ^^" 
> We should be able to do this without the circe-fs2 dependency

Ah I was hesitant to add it myself. Ought to have put more attention to what you said in Gitter.

Nice of you to give me the answer outright but I think I would've learned more if you gave me hints and I had to grind it out myself haha